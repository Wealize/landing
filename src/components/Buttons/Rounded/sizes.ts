import { ButtonSize } from '../../../interfaces/Buttons/ButtonSize'

export const SMALL: ButtonSize = {
  mobile: [90, 50],
  tablet: [150, 50],
  desktop: [170, 50],
  wide: [190, 50]
}

export const MEDIUM: ButtonSize = {
  mobile: [150, 70],
  tablet: [180, 80],
  desktop: [200, 85],
  wide: [220, 90]
}
