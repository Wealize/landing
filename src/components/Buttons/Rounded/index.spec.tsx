import React from 'react'
import { shallow } from 'enzyme'

import {
  ACCENT_COLOR,
  DARK_GRAY_COLOR,
  WHITE_COLOR
} from '../../../theme/color'

import ButtonRounded from './index'
import {
  Content
} from './styles'
import { SMALL } from './sizes'

describe('<ButtonRounded />', () => {
  const buttonProps = {
    backgroundColor: DARK_GRAY_COLOR,
    textColor: WHITE_COLOR,
    animatedBackgroundColor: ACCENT_COLOR,
    animatedTextColor: WHITE_COLOR,
    size: SMALL,
    children: <span>Hello</span>,
    handleClick: null
  }
  const wrapper = shallow(<ButtonRounded {...buttonProps} />)

  it('wrapper: should render', () => {
    expect(wrapper).toBeDefined()
    expect(wrapper).toMatchSnapshot()
  })

  it('wrapper: should render content correctly', () => {
    expect(wrapper.find(Content).first().text()).toEqual('Hello')
    expect(wrapper.find(Content).at(1).text()).toEqual('Hello')
  })
})
