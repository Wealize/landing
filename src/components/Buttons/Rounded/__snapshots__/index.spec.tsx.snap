// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonRounded /> wrapper: should render 1`] = `
<styled.div
  animatedBackgroundColor="#6666ff"
  animatedTextColor="#ffffff"
  backgroundColor="#3c3c3b"
  onClick={[Function]}
  size={
    Object {
      "desktop": Array [
        170,
        50,
      ],
      "mobile": Array [
        90,
        50,
      ],
      "tablet": Array [
        150,
        50,
      ],
      "wide": Array [
        190,
        50,
      ],
    }
  }
  textColor="#ffffff"
>
  <styled.button
    aria-label="Test text"
    backgroundColor="#3c3c3b"
  >
    <styled.span
      textColor="#ffffff"
    >
      Test text
    </styled.span>
  </styled.button>
  <styled.button
    animatedBackgroundColor="#6666ff"
    aria-label="Test text animated"
  >
    <styled.span
      textColor="#ffffff"
    >
      Test text
    </styled.span>
  </styled.button>
</styled.div>
`;
